### ISSUE TITLE
🐞 Bug report: at top of homepage: Deprecated: Function Elementor\DB::is_built_with_elementor is deprecated since version 3.2.0! Use Plugin::$instance->documents->get( $post_id )->is_built_with_elementor() instead. in /home1/geocen74/public_html/wp-includes/functions.php on line 5379

### ISSUE THREAD

**Issue by GeoCen** [elementor/elementor#18884]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Deprecated: Function Elementor\DB::is_built_with_elementor is deprecated since version 3.2.0! Use Plugin::$instance->documents->get( $post_id )->is_built_with_elementor() instead. in /home1/geocen74/public_html/wp-includes/functions.php on line 5379
shows up at top of homepage.  I tried all i can think of.

### Steps to reproduce

open homepage or editor

### Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

hp laptop and android tablet

**Comment by norazianti** [comment]
I am facing similar issue. Appreciate any help.

**Comment by rami-elementor** [comment]
Hi @GeoCen @norazianti  ,

One of your Elementor addons is using deprecated code.

You need to disable your plugin one-by-one until you find the one that breaks your site.

When you find the plugin / Elementor addon, you need to contact the addon developers and ask them to read about the [Elementor deprecation process](https://developers.elementor.com/docs/deprecations/) and update the addon code.

**Comment by axmedov2001** [comment]
**THE SOLUTOIN :** From left menu go to **Elementor / Tools  / Version Control** (Rollback Version) and reinstall the previous version.   

This may be due to the late update of the Esstentials for Elementor plugin in particular. In this way, you can fix the problem by installing elementor to the previous version.

**Comment by MarcoJHB** [comment]
@axmedov2001 this worked. I rolled back Elementor (not Pro) to 3.6, error seems to be gone now.

**Comment by getcordial** [comment]
Thank you for this. I installed 3.7.0 and it came up with the same error. Rolling back just Elementor fixed it, thanks @axmedov2001 
I am not using Essentials for Elementor. I also went though and disabled/enabled all plugins one at a time and the error remained. The only plugin I didn't disable was woocommerce

**Comment by aaartech** [comment]
Same issue here 
There is no separate plugin except Elementor + Pro on the site...

**Comment by DesignInvento** [comment]
use \Elementor\Plugin::$instance->documents->get( $post->ID )->is_built_with_elementor(); instead and make sure to not run this function in editor mode.

**Comment by amiriqbalmcs** [comment]
> Same issue here There is no separate plugin except Elementor + Pro on the site...

Yes, the same issue, even only on Elementor [Free version].

**Comment by axmedov2001** [comment]
> > Same issue here There is no separate plugin except Elementor + Pro on the site...
> 
> Yes, the same issue, even only on Elementor [Free version].

Maybe you can try change PHP version from server.

**Comment by rodpancine** [comment]
I solved it going to cPanel:
 /wp-content/plugins/elementor-pro/modules/theme-builder/documents/theme-document.php

Line 46 was

$method = $reflection->getMethod( 'get_site_editor_type' );

Changed it to

//$method = $reflection->getMethod( 'get_site_editor_type' );

**Comment by amiriqbalmcs** [comment]
> I solved it going to cPanel: /wp-content/plugins/elementor-pro/modules/theme-builder/documents/theme-document.php
> 
> Line 46 was
> 
> $method = $reflection->getMethod( 'get_site_editor_type' );
> 
> Changed it to
> 
> //$method = $reflection->getMethod( 'get_site_editor_type' );

The issue is in the Elementor Free Version

**Comment by axmedov2001** [comment]
> > I solved it going to cPanel: /wp-content/plugins/elementor-pro/modules/theme-builder/documents/theme-document.php
> > Line 46 was
> > $method = $reflection->getMethod( 'get_site_editor_type' );
> > Changed it to
> > //$method = $reflection->getMethod( 'get_site_editor_type' );
> 
> The issue is in the Elementor Free Version

What will happen when elementor is updated? It should be copied to the child theme folder, is it true?

**Comment by shilo-ey** [comment]
Hey,

Please make sure your theme is up to date and doesn't cause this issue.

If you find your theme as the origin of this error please contact your theme author and point them to this deprecation message.

Thanks.

**Comment by johnsonudofia** [comment]
I solved this error by rolling back my elementor version to Version 3.6.3 and also installing the HappyAddons plugin and the error was solved

**Comment by rami-elementor** [comment]
@rubencodingtel 

Please create an issue at Astra Github Repo https://github.com/brainstormforce/astra

**Comment by MianZahidN** [comment]
> In my case was it the Astra theme that triggered the error, I searched the following path ...wp-content/themes/astra/inc/compatibility/class-astra-elementor.php in Cpanel:
> 
> Astra Version: 3.9.1 Elementor free Version: 3.7.4
> 
> At line 262 was:
> 
> return Plugin::$instance->db->is_built_with_elementor( $id );
> 
> changed it to:
> 
> return Plugin::$instance->documents->get( $id )->is_built_with_elementor();
> 
> This solves my problem

This is also worked with me. Thank you!

**Comment by Pa1ad1n** [comment]
I am getting this same error. This all started when Elemenator updated itself then I had an error (nothing loaded at all) with that and the PHP version I used so I updated that.  The page loads but any time I try updating anything I get the server 500 error.

So I turned debugging on and now I get the error about Depreciated Function on the debug screen.

I have no plug ins active except Elemenator itself. I also had essentials add ons active but tried with it off as well.

I backed down the Elemenator version from 3.7.4 to 3.6.3 and still get the 500 error when trying to update.

Unsure what else to do.

Any suggestions?
